tfe/resource_tfe_workspace.go

```diff
@@ -15,6 +15,9 @@ func resourceTFEWorkspace() *schema.Resource {
 		Read:   resourceTFEWorkspaceRead,
 		Update: resourceTFEWorkspaceUpdate,
 		Delete: resourceTFEWorkspaceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
```

Comment:
The problem here seems to be that you explicitly (I'm guessing) set the `branch` to `master`. But the workspace is configured to use the default. This actually means `master`, but our API handles these defaults a bit strange.

If you create a workspace without a `branch`, it will use the default when executing, but it will not write that default as the configured value to the DB. So that means the actual value is an empty string.

So in this case if you omit the `branch` it works as expected, or if you apply this diff it will update the value in the DB to actually store `master`. That would not cause anything else to change as that is already the default being used, but it would fix the diff.

Of course I would suggest to just delete the `branch = "master"` from your config.

Reply:
I'll try removing `branch = "master"` and see if that removes this whole diff.